cmd/gardener-extension-shoot-dns-service/app/app.go

```diff
@@ -87,9 +88,10 @@ func (o *Options) run(ctx context.Context) error {
 	mgrOpts := o.managerOptions.Completed().Options()
 	mgrOpts.Scheme = mgrScheme
 	mgrOpts.ClientDisableCacheFor = []client.Object{
-		&corev1.Secret{},    // applied for ManagedResources
-		&corev1.ConfigMap{}, // applied for monitoring config
-		&dnsapi.DNSOwner{},  // avoid watching DNSOwner
+		&corev1.Secret{},      // applied for ManagedResources
+		&corev1.ConfigMap{},   // applied for monitoring config
+		&dnsapi.DNSOwner{},    // avoid watching DNSOwner
+		&v1alpha1.DNSRecord{}, // avoid watching DNSRecord
```

Comment:
Any reason why you don't want to cache `DNSRecords`? A watch cache is usually already created at the API server and the records aren't that big so that the memory footprint should be a problem. WDYT?

Reply:
There was no hard reason not to cache `DNSRecords`. I can activate it if you think it is the better choice.

Reply:
I think we can benefit from a cache here.